File: android/src/main/java/com/freeraspreactnative/Threat.kt
package com.freeraspreactnative

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableArray

/**
 * Sealed class to represent the error codes.
 *
 * Sealed classes are used because of obfuscation - enums classes are not obfuscated well enough.
 *
 * @property value integer value of the error code.
 */
internal sealed class Threat(val value: Int) {
  object AppIntegrity : Threat((10000..999999999).random())
  object PrivilegedAccess : Threat((10000..999999999).random())
  object Debug : Threat((10000..999999999).random())
  object Hooks : Threat((10000..999999999).random())
  object Passcode : Threat((10000..999999999).random())
  object Simulator : Threat((10000..999999999).random())
  object SecureHardwareNotAvailable : Threat((10000..999999999).random())
  object DeviceBinding : Threat((10000..999999999).random())
  object UnofficialStore : Threat((10000..999999999).random())
  object ObfuscationIssues : Threat((10000..999999999).random())
  object SystemVPN : Threat((10000..999999999).random())
  object DevMode : Threat((10000..999999999).random())
  object Malware : Threat((10000..999999999).random())
  object ADBEnabled : Threat((10000..999999999).random())
  object Screenshot : Threat((10000..999999999).random())
  object ScreenRecording : Threat((10000..999999999).random())

  companion object {
    internal fun getThreatValues(): WritableArray {
      return Arguments.fromList(
        listOf(
          AppIntegrity.value,
          PrivilegedAccess.value,
          Debug.value,
          Hooks.value,
          Passcode.value,
          Simulator.value,
          SecureHardwareNotAvailable.value,
          SystemVPN.value,
          DeviceBinding.value,
          UnofficialStore.value,
          ObfuscationIssues.value,
          DevMode.value,
          Malware.value,
          ADBEnabled.value,
          Screenshot.value,
          ScreenRecording.value
        )
      )
    }
  }
}

File: src/threat.ts
import { Platform } from 'react-native';

export class Threat {
  value: number;

  static AppIntegrity = new Threat(0);
  static PrivilegedAccess = new Threat(0);
  static Debug = new Threat(0);
  static Hooks = new Threat(0);
  static Passcode = new Threat(0);
  static Simulator = new Threat(0);
  static SecureHardwareNotAvailable = new Threat(0);
  static SystemVPN = new Threat(0);
  static DeviceBinding = new Threat(0);
  static DeviceID = new Threat(0);
  static UnofficialStore = new Threat(0);
  static ObfuscationIssues = new Threat(0);
  static DevMode = new Threat(0);
  static Malware = new Threat(0);
  static ADBEnabled = new Threat(0);
  static Screenshot = new Threat(0);
  static ScreenRecording = new Threat(0);

  constructor(value: number) {
    this.value = value;
  }

  static getValues(): Threat[] {
    return Platform.OS === 'android'
      ? [
          this.AppIntegrity,
          this.PrivilegedAccess,
          this.Debug,
          this.Hooks,
          this.Passcode,
          this.Simulator,
          this.SecureHardwareNotAvailable,
          this.SystemVPN,
          this.DeviceBinding,
          this.UnofficialStore,
          this.ObfuscationIssues,
          this.DevMode,
          this.Malware,
          this.ADBEnabled,
          this.Screenshot,
          this.ScreenRecording,
        ]
      : [
          this.AppIntegrity,
          this.PrivilegedAccess,
          this.Debug,
          this.Hooks,
          this.Passcode,
          this.Simulator,
          this.SecureHardwareNotAvailable,
          this.SystemVPN,
          this.DeviceBinding,
          this.DeviceID,
          this.UnofficialStore,
          this.Screenshot,
          this.ScreenRecording,
        ];
  }
}
